### ISSUE TITLE
Which modules to implement from the Python standard library

### ISSUE THREAD

**Issue by certik** [lcompilers/lpython#200]
The Python standard library has a lot of modules: https://docs.python.org/3/library/index.html, but many of them are rarely used. First we should concentrate on the most commonly used modules, probably:

* `sys`
* `os`
* `string`
* `datetime`
* `cmath`
* `decimal`
* `numbers`
* `fractions`
* `statistics`

**Comment by Smit-create** [comment]
I can work on implementing `statistics`.

**Comment by namannimmo10** [comment]
I can do the `random` module

**Comment by Oshanath** [comment]
Should these modules be implemented in C++ or in LPython itself?

**Comment by Oshanath** [comment]
Basically if I were to implement something from this list, what I should do is refer to the Python documentation of that module and implement those same functions the same way using LPython. Am I correct?

**Comment by certik** [comment]
Yes, and add them to `src/runtime` and add tests to `integration_tests` and register them in `integration_tests/run_tests.py`, they will be run via CPython (to ensure the test script is written correctly) as well as LPython (to ensure everything actually compiles).

**Comment by rajpratyush** [comment]
I think itertools, datetime, sys and os should also be added to the lpython since they are used extensively in scientific computing. i can start the work for sys and os modules.

**Comment by namannimmo10** [comment]
> I think itertools, datetime, sys and os should also be added to the lpython since they are used extensively in scientific computing. i can start the work for sys and os modules.

Great. Let us know if you need any help!

**Comment by rajpratyush** [comment]
sure could you provide me up links to the development guide so that I can set up the project locally. or any kind of documentation website

**Comment by namannimmo10** [comment]
You can find the installation guide in the README.md file.  For now, see https://docs.lfortran.org/developer_tutorial/ for more information -- LPython follows the same design as in LFortran.

**Comment by rajpratyush** [comment]
so it doesn't have a documentation website ?

**Comment by certik** [comment]
We currently don't have dedicated LPython documentation. We will eventually add it.

**Comment by rajpratyush** [comment]
how about building one?

**Comment by certik** [comment]
Yes, we should.

**Comment by Umesh-JNU** [comment]
I would like to work for argparse. Can you please help me how to start?

**Comment by kmr-srbh** [comment]
Hello! I got to know about LPython through the GSoC page of Python. I am interested in helping LPython develop these modules for science. I have a strong command over Python and a keen interest in programming languages.

Ever since I got to know about type-hinting in Python, I fell in love with the idea. I have been using it extensively in my code since then. Seeing LPython code look the same, I just loved it!

I went through the tutorial and saw code inside `src/runtime`. I would like some guidance from you on getting started with writing code. I feel comfortable with the programming language(LPython) and the method stated above for going about the development .i.e. developing the modules according to the documentation by PSF.

When I came to the repository, I too searched for the documentation. I would love to start work on it along with adding more tutorials.

The easy Python syntax coupled with the speed of a compiled language is a great combination! 
Together, we will make it big!
